### PR TITLE
Add --class-const-naming-style

### LINKED ISSUE #4181: Make it possible to allow lower case enum names
### Is your feature request related to a problem? Please describe
With #3834 enums are now required to be in all caps, and seem to be treated equal to normal constants. However, existing code bases can't simply change all enum values, as this backwards incompatible, and adding an ignore to each member is a lot of noise.

### Describe the solution you'd like
Add a new regex or option to change the allowed way enum values can look

### PR DESCRIPTION
**Pull request by cdce8p**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Add a new naming style `class-const`.
This will apply to:
- Enum constants
- Class variables if annotated with `ClassVar`

Although this technically is a new feature, there might be a point in including it in `2.7.3` (see issue below).

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |
| ✓  | :sparkles: New feature |
|   | :hammer: Refactoring  |
| ✓  | :scroll: Docs |

## Related Issue
Closes #4181

### CODE REVIEW

File: pylint/checkers/base.py [hunk 1]
```diff
@@ -1983,8 +1988,18 @@ def visit_assignname(self, node):
         elif isinstance(frame, astroid.ClassDef):
             if not list(frame.local_attr_ancestors(node.name)):
                 for ancestor in frame.ancestors():
-                    if ancestor.name == "Enum" and ancestor.root().name == "enum":
-                        self._check_name("const", node.name, node)
+                    if (  # pylint: disable=too-many-boolean-expressions
+                        ancestor.name == "Enum"
+                        and ancestor.root().name == "enum"
+                        or isinstance(node.parent, astroid.AnnAssign)
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/base.py:1994]: Maybe we can create an ``is_class_var`` function and remove the disable ?

**Review by Pierre-Sassoulas — APPROVED** [review]
Nice feature for 2.8 :)

**Comment by cdce8p** [comment]
A `utils.is_class_var` function does make a lot of sense. I updated the MR.

I would consider it a fix. The impact is most likely limited and can be easily mitigated by adding `class-const-naming-style=any` to the `pylintrc` file.

**Comment by cdce8p** [comment]
I believe this MR is ready for merging

**Pull request merged**